api/worker/api.go

```diff
@@ -2,17 +2,26 @@ package worker
 
 import (
 	"github.com/gorilla/mux"
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker/auth"
+	"github.com/ufcg-lsd/arrebol-pb/arrebol/worker/manager"
+	"github.com/ufcg-lsd/arrebol-pb/storage"
 	"log"
 	"net/http"
 )
 
 type WorkerApi struct {
 	server  *http.Server
+	manager manager.Manager
+	auth auth.Auth
```

Comment:
let's call it authenticator, ok? Also, let's try to rename its functions to something more meaningful, for example: What a func called Authenticate should do is to guarantee that the worker is who it is saying it is. The way we verify that is by checking if its signature can be verified with its public key, right? It seems that the Validate func is doing this... We may be mixing concepts here... It is an example, I suggest to rethink all funcs from this module

Reply:
they're improving, nice :)